// @ssheverev/eslint-config/src/configs/base.js
import resetBase from "../reset/base.js";

export default [
  resetBase,
  {
    ignores: [
      // dependencies
      "**/node_modules/**",
      ".pnp.*",

      // build outputs
      "**/dist/**",
      "**/build/**",
      "**/out/**",
      "**/.next/**",
      "**/coverage/**",

      // cache
      ".eslintcache",
      ".cache",
      ".npm",

      // misc
      "**/.DS_Store",
      "*.pem",
      ".env*",

      // debug
      "npm-debug.log*",
      "yarn-debug.log*",
      "yarn-error.log*",

      // IDE
      ".idea",
      ".vscode/*",
      "!.vscode/settings.json",
      "!.vscode/tasks.json",
      "!.vscode/launch.json",
      "!.vscode/extensions.json",
    ],
    languageOptions: {
      parser: undefined,
      parserOptions: {
        ecmaVersion: "latest",
        sourceType: "module",
        ecmaFeatures: {
          jsx: true,
        },
      },
    },
    files: ["**/*.js", "**/*.jsx"],
    rules: {
      "no-console": ["warn", { allow: ["error", "info", "warn"] }],
      "dot-notation": "warn",
      "no-implied-eval": "error",
      "no-unused-vars": ["warn", { ignoreRestSiblings: true }],
      "no-use-before-define": "warn",
    },
  },
];
