// @ssheverev/eslint-config/src/configs/stylistic.js
import resetStylistic from "../reset/stylistic.js";

export default [
  resetStylistic,
  {
    rules: {
      "stylistic/arrow-spacing": ["warn", { before: true, after: true }], // есть в prettier
      "stylistic/brace-style": "warn", // есть в prettier
      "stylistic/comma-spacing": ["warn", { before: false, after: true }], // есть в prettier
      "stylistic/computed-property-spacing": ["warn", "never"], // есть в prettier
      "stylistic/dot-location": ["warn", "property"],
      "stylistic/eol-last": "warn", // есть в prettier
      "stylistic/function-call-argument-newline": ["error", "consistent"],
      "stylistic/function-call-spacing": ["error", "never"], // есть в prettier
      "stylistic/generator-star-spacing": [
        "warn",
        { before: false, after: true },
      ], // есть в prettier
      "stylistic/jsx-quotes": "error",
      "stylistic/key-spacing": "warn", // есть в prettier
      "stylistic/keyword-spacing": "warn", // есть в prettier
      "stylistic/lines-around-comment": ["warn", { beforeBlockComment: true }],
      "stylistic/lines-between-class-members": ["warn", "always"],
      "stylistic/max-statements-per-line": "warn",
      "stylistic/member-delimiter-style": "warn",
      "stylistic/new-parens": "error",
      "stylistic/no-floating-decimal": "warn",
      "stylistic/no-mixed-spaces-and-tabs": "warn", // есть в prettier
      "stylistic/no-multi-spaces": "warn", // есть в prettier
      "stylistic/no-multiple-empty-lines": [
        "warn",
        {
          max: 1,
          maxEOF: 1,
          maxBOF: 0,
        },
      ],
      "stylistic/no-trailing-spaces": ["warn", { ignoreComments: true }], // есть в prettier
      "stylistic/no-whitespace-before-property": "warn", // есть в prettier
      "stylistic/object-curly-newline": ["warn", { consistent: true }], // есть в prettier
      "stylistic/padded-blocks": ["warn", "never"], // есть в prettier
      "stylistic/padding-line-between-statements": [
        "warn",
        {
          blankLine: "always",
          prev: "*",
          next: "return",
        },
        {
          blankLine: "always",
          prev: "*",
          next: "if",
        },
      ],
      "stylistic/rest-spread-spacing": ["warn", "never"], // есть в prettier
      "stylistic/semi": ["warn", "always"], // есть в prettier
      "stylistic/space-in-parens": ["warn", "never"], // есть в prettier
      "stylistic/space-infix-ops": "warn", // есть в prettier
      "stylistic/space-unary-ops": ["warn", { words: true, nonwords: false }], // есть в prettier
      "stylistic/spaced-comment": "warn",
      "stylistic/switch-colon-spacing": "warn",
      "stylistic/template-curly-spacing": "warn", // есть в prettier
      "stylistic/type-annotation-spacing": "warn", // есть в prettier
      "stylistic/type-generic-spacing": "warn", // есть в prettier
      "stylistic/type-named-tuple-spacing": "warn", // есть в prettier
      "stylistic/wrap-iife": [
        "warn",
        "inside",
        { functionPrototypeMethods: true },
      ], // есть в prettier
      "stylistic/yield-star-spacing": ["warn", "after"],
    },
  },
];
