// @ssheverev/eslint-react-config/src/configs/react.js
import resetReact from '../reset/react.js';

export default [
  resetReact,
  {
    settings: {
      react: {
        version: 'detect'
      },
    },
    rules: {
      'react/forward-ref-uses-ref': 'error',
      'react/hook-use-state': 'warn',
      'react/jsx-boolean-value': ['error', 'never'],
      'react/jsx-closing-bracket-location': ['error', 'tag-aligned'],
      'react/jsx-closing-tag-location': ['error', 'tag-aligned'],
      'react/jsx-curly-brace-presence': 'error',
      'react/jsx-curly-spacing': 'warn',
      'react/jsx-equals-spacing': 'warn',
      'react/jsx-key': 'error',
      'react/jsx-first-prop-new-line': ['warn', 'multiline'],
      'react/jsx-max-props-per-line': ['warn', {maximum: 1, when: 'multiline'}],
      'react/jsx-no-constructed-context-values': 'warn',
      'react/jsx-no-duplicate-props': 'error',
      'react/jsx-pascal-case': 'error',
      'react/jsx-props-no-multi-spaces': 'error',
      'react/jsx-sort-props': ['warn', {callbacksLast: true}],
      'react/jsx-tag-spacing': ['warn', {beforeSelfClosing: 'always'}],
      'react/jsx-uses-react': 'warn',
      'react/jsx-uses-vars': 'error',
      'react/jsx-wrap-multilines': [
        'warn',
        {
          declaration: 'parens-new-line',
          assignment: 'parens-new-line',
          return: 'parens-new-line',
          arrow: 'parens-new-line',
          condition: 'parens-new-line',
          logical: 'parens-new-line',
          prop: 'ignore',
        },
      ],
      'react/no-access-state-in-setstate': 'error',
      'react/no-children-prop': 'error',
      'react/no-did-mount-set-state': 'error',
      'react/no-did-update-set-state': 'error',
      'react/no-direct-mutation-state': 'error',
      'react/no-find-dom-node': 'error',
      'react/no-is-mounted': 'error',
      'react/no-multi-comp': 'warn', // ?
      'react/no-render-return-value': 'error',
      'react/no-string-refs': 'error',
      'react/require-render-return': 'error',
      'react/self-closing-comp': 'warn',
      'react/style-prop-object': 'error',
    },
  },
];
