// @ssheverev/prettier-config/index.js
export default {
  printWidth: 120, // определяет максимальную длину строки, после которой происходит перенос
  tabWidth: 4, // определяет количество пробелов для одного уровня отступа
  arrowParens: 'always', // определяет, нужно ли всегда использовать скобки вокруг параметров стрелочных функций
  useTabs: true, // определяет, использовать ли символы табуляции вместо пробелов для отступов
  semi: true, // определяет, нужно ли добавлять точку с запятой в конце выражений
  singleQuote: true, // определяет, использовать ли одинарные или двойные кавычки для строк
  jsxSingleQuote: false, // определяет, какие кавычки использовать для атрибутов в JSX
  quoteProps: 'as-needed', // определяет, когда нужно заключать ключи объектов в кавычки
  trailingComma: 'all', // когда добавлять замыкающие запятые в многострочных конструкциях (объектах, массивах, параметрах функций)
  bracketSpacing: false, // контролирует наличие пробелов внутри фигурных скобок объектов
  bracketSameLine: false, // определяет, где должна располагаться закрывающая скобка в многострочных JSX элементах и HTML тегах
  endOfLine: 'lf', // определяет тип символа конца строки в файлах
  embeddedLanguageFormatting: 'auto', // определяет, как форматировать код внутри шаблонных строк или других встроенных языков
  singleAttributePerLine: true, // определяет, нужно ли размещать каждый атрибут JSX/HTML на отдельной строке
  proseWrap: 'always', // определяет, как форматировать длинные текстовые строки в markdown файлах и комментариях
};
