// @ssheverev/eslint-react-config/src/reset/react.js
import react from 'eslint-plugin-react';

export default {
  plugins: {
    react,
  },
  rules: {
    // Description: Enforces consistent naming for boolean props
    // Description RU: Обеспечивает согласованное присвоение имен логическим элементам props
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/boolean-prop-naming.md
    'react/boolean-prop-naming': 'off',

    // Description: Disallow usage of button elements without an explicit type attribute
    // Description RU: Запрещает использование элементов button без явного атрибута type
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/button-has-type.md
    'react/button-has-type': 'off',

    // Description: Enforce using onChange or readonly attribute when checked is used
    // Description RU: Принудительно использовать атрибут onChange или readonly, если установлен флажок
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/checked-requires-onchange-or-readonly.md
    'react/checked-requires-onchange-or-readonly': 'off',

    // Description: Enforce all defaultProps have a corresponding non-required PropType
    // Description RU: Для обеспечения того, чтобы все программы defaultProps имели соответствующий необязательный тип propType
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/default-props-match-prop-types.md
    'react/default-props-match-prop-types': 'off',

    // Description: Enforce consistent usage of destructuring assignment of props, state, and context
    // Description RU: Обеспечивает согласованное использование деструктурирующего назначения реквизитов, состояния и контекста
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/destructuring-assignment.md
    'react/destructuring-assignment': 'off',

    // Description: Disallow missing displayName in a React component definition
    // Description RU: Запрещает пропуск отображаемого имени в определении компонента React
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/display-name.md
    'react/display-name': 'off',

    // Description: Disallow certain props on components
    // Description RU: Запрещает определенные настройки в компонентах
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/forbid-component-props.md
    'react/forbid-component-props': 'off',

    // Description: Disallow certain props on DOM Nodes
    // Description RU: Запрещает определенные настройки в узлах DOM
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/forbid-dom-props.md
    'react/forbid-dom-props': 'off',

    // Description: Disallow certain elements
    // Description RU: Запрещает использование определенных элементов
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/forbid-elements.md
    'react/forbid-elements': 'off',

    // Description: Disallow using another component's propTypes
    // Description RU: Запретить использование свойств другого компонента
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/forbid-foreign-prop-types.md
    'react/forbid-foreign-prop-types': 'off',

    // Description: Disallow certain propTypes
    // Description RU: Запретить определенные свойства
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/forbid-prop-types.md
    'react/forbid-prop-types': 'off',

    // Description: Require all forwardRef components include a ref parameter
    // Description RU: Требуется, чтобы все компоненты forwardRef включали параметр ref
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/forward-ref-uses-ref.md
    'react/forward-ref-uses-ref': 'off',

    // Description: Enforce a specific function type for function components
    // Description RU: Принудительно использовать определенный тип функции для функциональных компонентов
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/function-component-definition.md
    'react/function-component-definition': 'off',

    // Description: Ensure destructuring and symmetric naming of useState hook value and setter variables
    // Description RU: Обеспечьте деструктуризацию и симметричное присвоение имен значению перехвата useState и переменным-установщикам
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/hook-use-state.md
    'react/hook-use-state': 'off',

    // Description: Enforce sandbox attribute on iframe elements
    // Description RU: Примените атрибут sandbox к элементам iframe
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/iframe-missing-sandbox.md
    'react/iframe-missing-sandbox': 'off',

    // Description: Enforce boolean attributes notation in JSX
    // Description RU: Принудительное использование логических атрибутов в JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-boolean-value.md
    'react/jsx-boolean-value': 'off',

    // Description: Enforce or disallow spaces inside of curly braces in JSX attributes and expressions
    // Description RU: Использование или запрет пробелов внутри фигурных скобок в атрибутах и выражениях JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-child-element-spacing.md
    'react/jsx-child-element-spacing': 'off',

    // Description: Enforce closing bracket location in JSX
    // Description RU: Принудительное расположение закрывающей скобки в JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-closing-bracket-location.md
    'react/jsx-closing-bracket-location': 'off',

    // Description: Enforce closing tag location for multiline JSX
    // Description RU: Принудительное расположение закрывающего тега для многострочного JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-closing-tag-location.md
    'react/jsx-closing-tag-location': 'off',

    // Description: Disallow unnecessary JSX expressions when literals alone are sufficient or enforce JSX expressions on literals in JSX children or attributes
    // Description RU: Запрещает использование ненужных выражений JSX, когда достаточно одних литералов, или применяет выражения JSX к литералам в дочерних элементах или атрибутах JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-curly-brace-presence.md
    'react/jsx-curly-brace-presence': 'off',

    // Description: Enforce consistent linebreaks in curly braces in JSX attributes and expressions
    // Description RU:  Принудительное использование последовательных переносов строк в фигурных скобках в атрибутах и выражениях JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-curly-newline.md
    'react/jsx-curly-newline': 'off',

    // Description: Enforce or disallow spaces inside of curly braces in JSX attributes and expressions
    // Description RU: Применение или запрет пробелов внутри фигурных скобок в атрибутах и выражениях JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-curly-spacing.md
    'react/jsx-curly-spacing': 'off',

    // Description: Enforce or disallow spaces around equal signs in JSX attributes
    // Description RU: Применение или запрет пробелов вокруг знаков равенства в атрибутах JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-equals-spacing.md
    'react/jsx-equals-spacing': 'off',

    // Description: Disallow file extensions that may contain JSX
    // Description RU: Запрещает использование расширений файлов, которые могут содержать JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-filename-extension.md
    'react/jsx-filename-extension': 'off',

    // Description: Enforce proper position of the first property in JSX
    // Description RU: Необходимо обеспечить правильное расположение первого свойства в JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-first-prop-new-line.md
    'react/jsx-first-prop-new-line': 'off',

    // Description: Enforce shorthand or standard form for React fragments
    // Description RU: Применение сокращенной или стандартной формы для фрагментов React
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-fragments.md
    'react/jsx-fragments': 'off',

    // Description: Enforce event handler naming conventions in JSX
    // Description RU: Применение соглашений об именовании обработчиков событий в JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-handler-names.md
    'react/jsx-handler-names': 'off',

    // Description: Enforce JSX indentation
    // Description RU: Принудительное использование отступов в JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-indent.md
    'react/jsx-indent': 'off',

    // Description: Enforce props indentation in JSX
    // Description RU: Принудительное использование реквизитного отступа в JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-indent-props.md
    'react/jsx-indent-props': 'off',

    // Description: Disallow missing key props in iterators/collection literals
    // Description RU: Запретить использование отсутствующих реквизитов ключей в итераторах/литералах коллекций
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-key.md
    'react/jsx-key': 'off',

    // Description: Enforce JSX maximum depth
    // Description RU: Обеспечить максимальную глубину JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-max-depth.md
    'react/jsx-max-depth': 'off',

    // Description: Enforce maximum of props on a single line in JSX
    // Description RU: Принудительное использование максимального количества реквизитов в одной строке в JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-max-props-per-line.md
    'react/jsx-max-props-per-line': 'off',

    // Description: Require or prevent a new line after jsx elements and expressions.
    // Description RU: Требует или запрещает создание новой строки после элементов и выражений jsx.
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-newline.md
    'react/jsx-newline': 'off',

    // Description: Disallow .bind() or arrow functions in JSX props
    // Description RU: Запрещает .bind() или функции со стрелками в JSX props
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-no-bind.md
    'react/jsx-no-bind': 'off',

    // Description: Disallow comments from being inserted as text nodes
    // Description RU: Запретить вставку комментариев в виде текстовых узлов
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-no-comment-textnodes.md
    'react/jsx-no-comment-textnodes': 'off',

    // Description: Disallows JSX context provider values from taking values that will cause needless rerenders
    // Description RU: Не позволяет поставщику контекстных данных JSX принимать значения, которые приведут к ненужным повторным отправкам
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-no-constructed-context-values.md
    'react/jsx-no-constructed-context-values': 'off',

    // Description: Disallow duplicate properties in JSX
    // Description RU: Запрещает дублирование свойств в JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-no-duplicate-props.md
    'react/jsx-no-duplicate-props': 'off',

    // Description: Disallow problematic leaked values from being rendered
    // Description RU: Запрещает отображение проблемных пропущенных значений
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-no-leaked-render.md
    'react/jsx-no-leaked-render': 'off',

    // Description: Disallow usage of string literals in JSX
    // Description RU: Запрещает использование строковых литералов в JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-no-literals.md
    'react/jsx-no-literals': 'off',

    // Description: Disallow usage of javascript: URLs
    // Description RU: Запретить использование javascript: URLs
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-no-script-url.md
    'react/jsx-no-script-url': 'off',

    // Description: Disallow target='_blank' attribute without rel='noreferrer'
    // Description RU: Запретить использование атрибута target='_blank' без атрибута rel='noreferrer'.
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-no-target-blank.md
    'react/jsx-no-target-blank': 'off',

    // Description: Disallow undeclared variables in JSX
    // Description RU: Запрещает использование необъявленных переменных в JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-no-undef.md
    'react/jsx-no-undef': 'off',

    // Description: Disallow unnecessary fragments
    // Description RU: Запрещает использование ненужных фрагментов
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-no-useless-fragment.md
    'react/jsx-no-useless-fragment': 'off',

    // Description: Require one JSX element per line
    // Description RU: Требуется один элемент JSX в строке
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-one-expression-per-line.md
    'react/jsx-one-expression-per-line': 'off',

    // Description: Enforce PascalCase for user-defined JSX components
    // Description RU: Принудительное использование PascalCase для пользовательских компонентов JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-pascal-case.md
    'react/jsx-pascal-case': 'off',

    // Description: Disallow multiple spaces between inline JSX props
    // Description RU: Запрещает использование нескольких пробелов между встроенными реквизитами JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-props-no-multi-spaces.md
    'react/jsx-props-no-multi-spaces': 'off',

    // Description: Disallow JSX prop spreading the same identifier multiple times
    // Description RU: Запретить распространение JSX prop с одним и тем же идентификатором несколько раз
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-props-no-spread-multi.md
    'react/jsx-props-no-spread-multi': 'off',

    // Description: Disallow JSX prop spreading
    // Description RU: Запретить распространение JSX prop
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-props-no-spreading.md
    'react/jsx-props-no-spreading': 'off',

    // Description: Enforce defaultProps declarations alphabetical sorting
    // Description RU: Принудительная сортировка по алфавиту в объявлениях defaultProps
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-sort-default-props.md
    'react/jsx-sort-default-props': 'off',

    // Description: Enforce props alphabetical sorting
    // Description RU: Принудительная сортировка реквизитов по алфавиту
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-sort-props.md
    'react/jsx-sort-props': 'off',

    // Description: Enforce spacing before closing bracket in JSX
    // Description RU: Принудительный интервал перед закрывающей скобкой в JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-space-before-closing.md
    'react/jsx-space-before-closing': 'off',

    // Description: Enforce whitespace in and around the JSX opening and closing brackets
    // Description RU: Принудительное использование пробелов в открывающих и закрывающих скобках JSX и вокруг них
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-tag-spacing.md
    'react/jsx-tag-spacing': 'off',

    // Description: Disallow React to be incorrectly marked as unused
    // Description RU: Запретить, чтобы React был неправильно помечен как неиспользуемый
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-uses-react.md
    'react/jsx-uses-react': 'off',

    // Description: Disallow variables used in JSX to be incorrectly marked as unused
    // Description RU: Запретить помечать React как неиспользуемый, если он неправильно используется
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-uses-vars.md
    'react/jsx-uses-vars': 'off',

    // Description: Disallow missing parentheses around multiline JSX
    // Description RU: Не допускайте, чтобы переменные, используемые в JSX, неправильно помечались как неиспользуемые
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/jsx-wrap-multilines.md
    'react/jsx-wrap-multilines': 'off',

    // Description: Disallow when this.state is accessed within setState
    // Description RU: Запретить при доступе к этому.state в setState
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-access-state-in-setstate.md
    'react/no-access-state-in-setstate': 'off',

    // Description: Disallow adjacent inline elements not separated by whitespace.
    // Description RU: Запрещает использование смежных встроенных элементов, не разделенных пробелом.
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-adjacent-inline-elements.md
    'react/no-adjacent-inline-elements': 'off',

    // Description: Disallow usage of Array index in keys
    // Description RU: Запрещает использование индекса массива в ключах
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-array-index-key.md
    'react/no-array-index-key': 'off',

    // Description: Lifecycle methods should be methods on the prototype, not class fields
    // Description RU: Методы жизненного цикла должны быть методами прототипа, а не полями класса
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-arrow-function-lifecycle.md
    'react/no-arrow-function-lifecycle': 'off',

    // Description: Disallow passing of children as props
    // Description RU: Запрещать пронос детей в качестве реквизита
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-children-prop.md
    'react/no-children-prop': 'off',

    // Description: Disallow usage of dangerous JSX properties
    // Description RU: Запрещает использование опасных свойств JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-danger.md
    'react/no-danger': 'off',

    // Description: Disallow when a DOM element is using both children and dangerouslySetInnerHTML
    // Description RU: Запрещает, когда DOM-элемент использует как дочерние элементы, так и опасный параметр innerHTML
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-danger-with-children.md
    'react/no-danger-with-children': 'off',

    // Description: Disallow usage of deprecated methods
    // Description RU: Запрещает использование устаревших методов
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-deprecated.md
    'react/no-deprecated': 'off',

    // Description: Disallow usage of setState in componentDidMount
    // Description RU: Запретить использование setState в componentDidMount
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-did-mount-set-state.md
    'react/no-did-mount-set-state': 'off',

    // Description: Disallow usage of setState in componentDidUpdate
    // Description RU: Запретить использование параметра setState в componentDidUpdate
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-did-update-set-state.md
    'react/no-did-update-set-state': 'off',

    // Description: Disallow direct mutation of this.state
    // Description RU: Запретить прямое изменение этого параметра.state
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-direct-mutation-state.md
    'react/no-direct-mutation-state': 'off',

    // Description: Disallow usage of findDOMNode
    // Description RU: Запрещает использование findDOMNode
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-find-dom-node.md
    'react/no-find-dom-node': 'off',

    // Description: Disallow usage of invalid attributes
    // Description RU: Запрещает использование недопустимых атрибутов
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-invalid-html-attribute.md
    'react/no-invalid-html-attribute': 'off',

    // Description: Disallow usage of isMounted
    // Description RU: Запретить использование isMounted
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-is-mounted.md
    'react/no-is-mounted': 'off',

    // Description: Disallow multiple component definition per file
    // Description RU: Запрещает определение нескольких компонентов для каждого файла
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-multi-comp.md
    'react/no-multi-comp': 'off',

    // Description: Enforce that namespaces are not used in React elements
    // Description RU: Убедитесь, что в элементах React не используются пространства имен
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-namespace.md
    'react/no-namespace': 'off',

    // Description: Disallow usage of referential-type variables as default param in functional component
    // Description RU: Запретите использование переменных ссылочного типа в качестве параметров по умолчанию в функциональном компоненте
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-object-type-as-default-prop.md
    'react/no-object-type-as-default-prop': 'off',

    // Description: Disallow usage of shouldComponentUpdate when extending React.PureComponent
    // Description RU: Запретить использование shouldComponentUpdate при расширении React.PureComponent
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-redundant-should-component-update.md
    'react/no-redundant-should-component-update': 'off',

    // Description: Disallow usage of the return value of ReactDOM.render
    // Description RU: Запретить использование возвращаемого значения ReactDOM.render
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-render-return-value.md
    'react/no-render-return-value': 'off',

    // Description: Disallow usage of setState
    // Description RU: Запретить использование setState
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-set-state.md
    'react/no-set-state': 'off',

    // Description: Disallow using string references
    // Description RU: Запрещает использование ссылок на строки
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-string-refs.md
    'react/no-string-refs': 'off',

    // Description: Disallow this from being used in stateless functional components
    // Description RU: Запрещает использование этого параметра в функциональных компонентах без сохранения состояния
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-this-in-sfc.md
    'react/no-this-in-sfc': 'off',

    // Description: Disallow common typos
    // Description RU: Запрещает типичные опечатки
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-typos.md
    'react/no-typos': 'off',

    // Description: Disallow unescaped HTML entities from appearing in markup
    // Description RU: Запрещает отображение неэкранированных HTML-объектов в разметке
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-unescaped-entities.md
    'react/no-unescaped-entities': 'off',

    // Description: Disallow usage of unknown DOM property
    // Description RU: Запретить использование неизвестного свойства DOM
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-unknown-property.md
    'react/no-unknown-property': 'off',

    // Description: Disallow usage of unsafe lifecycle methods
    // Description RU: Запретить использование небезопасных методов жизненного цикла
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-unsafe.md
    'react/no-unsafe': 'off',

    // Description: Disallow creating unstable components inside components
    // Description RU: Запрещает создание нестабильных компонентов внутри компонентов
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-unstable-nested-components.md
    'react/no-unstable-nested-components': 'off',

    // Description: Disallow declaring unused methods of component class
    // Description RU: Запрещает объявлять неиспользуемые методы класса component
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-unused-class-component-methods.md
    'react/no-unused-class-component-methods': 'off',

    // Description: Disallow definitions of unused propTypes
    // Description RU: Запретить определения неиспользуемых типов свойств
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-unused-prop-types.md
    'react/no-unused-prop-types': 'off',

    // Description: Disallow definitions of unused state
    // Description RU: Запретить определения неиспользуемого состояния
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-unused-state.md
    'react/no-unused-state': 'off',

    // Description: Disallow usage of setState in componentWillUpdate
    // Description RU: Запрещает использование setState в componentWillUpdate
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/no-will-update-set-state.md
    'react/no-will-update-set-state': 'off',

    // Description: Enforce ES5 or ES6 class for React Components
    // Description RU: Принудительное использование классов ES5 или ES6 для компонентов React
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/prefer-es6-class.md
    'react/prefer-es6-class': 'off',

    // Description: Prefer exact proptype definitions
    // Description RU: Предпочитайте точные определения типов proptype
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/prefer-exact-props.md
    'react/prefer-exact-props': 'off',

    // Description: Enforce that props are read-only
    // Description RU: Убедитесь, что props доступен только для чтения
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/prefer-read-only-props.md
    'react/prefer-read-only-props': 'off',

    // Description: Enforce stateless components to be written as a pure function
    // Description RU: Принудительно записывать компоненты без сохранения состояния как чистую функцию
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/prefer-stateless-function.md
    'react/prefer-stateless-function': 'off',

    // Description: Disallow missing props validation in a React component definition
    // Description RU: Запрещает проверку отсутствия props в определении компонента React
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/prop-types.md
    'react/prop-types': 'off',

    // Description: Disallow missing React when using JSX
    // Description RU: Запрещает проверку отсутствия React при использовании JSX
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/react-in-jsx-scope.md
    'react/react-in-jsx-scope': 'off',

    // Description: Enforce a defaultProps definition for every prop that is not a required prop
    // Description RU: Принудительно применять определение defaultProps для каждого prop, который не является обязательным prop
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/require-default-props.md
    'react/require-default-props': 'off',

    // Description: Enforce React components to have a shouldComponentUpdate method
    // Description RU: Принудительно применяем к компонентам React метод shouldComponentUpdate
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/require-optimization.md
    'react/require-optimization': 'off',

    // Description: Enforce ES5 or ES6 class for returning value in render function
    // Description RU: Принудительно применяем класс ES5 или ES6 для возврата значения в функции рендеринга
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/require-render-return.md
    'react/require-render-return': 'off',

    // Description: Disallow extra closing tags for components without children
    // Description RU: Запрещает использование дополнительных закрывающих тегов для компонентов без дочерних элементов
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/self-closing-comp.md
    'react/self-closing-comp': 'off',

    // Description: Enforce component methods order
    // Description RU: Обеспечивает соблюдение порядка использования методов
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/sort-comp.md
    'react/sort-comp': 'off',

    // Description: Enforce defaultProps declarations alphabetical sorting
    // Description RU: Принудительная сортировка объявлений defaultProps по алфавиту
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/sort-default-props.md
    'react/sort-default-props': 'off',

    // Description: Enforce propTypes declarations alphabetical sorting
    // Description RU: Принудительная сортировка по алфавиту в объявлениях PropTypes
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/sort-prop-types.md
    'react/sort-prop-types': 'off',

    // Description: Enforce class component state initialization style
    // Description RU: Применяет стиль инициализации состояния компонента класса
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/state-in-constructor.md
    'react/state-in-constructor': 'off',

    // Description: Enforces where React component static properties should be positioned.
    // Description RU: Определяет, где должны располагаться статические свойства компонента React.
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/static-property-placement.md
    'react/static-property-placement': 'off',

    // Description: Enforce style prop value is an object
    // Description RU: Принудительное использование параметра style - это объект
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/style-prop-object.md
    'react/style-prop-object': 'off',

    // Description: Disallow void DOM elements (e.g. <img />, <br />) from receiving children
    // Description RU: Запретить элементам void DOM (например, <img />, <br />) получать дочерние элементы
    // URL:https://github.com/jsx-eslint/eslint-plugin-react/blob/HEAD/docs/rules/void-dom-elements-no-children.md
    'react/void-dom-elements-no-children': 'off',
},
};
