// @ssheverev/eslint-config/src/reset/stylistic.js
import stylistic from '@stylistic/eslint-plugin'

export default {
  /*
   * Сброс всех правил в состояние "Нет ошибки". Для плавной интеграции
   */
  plugins: {
    stylistic
  },
  rules: {
    // Description: Enforce linebreaks after opening and before closing array brackets
    // Description RU: Принудительный перенос строк после открытия и перед закрытием скобок массива
    // URL: https://eslint.style/rules/default/array-bracket-newline
    'stylistic/array-bracket-newline': 'off',

    // Description: Enforce consistent spacing inside array brackets
    // Description RU: Принудительный согласованный интервал внутри скобок массива
    // URL: https://eslint.style/rules/default/array-bracket-spacing
    'stylistic/array-bracket-spacing': 'off',

    // Description: Enforce line breaks after each array element
    // Description RU: Принудительно использовать разрывы строк после каждого элемента массива
    // URL: https://eslint.style/rules/default/array-element-newline
    'stylistic/array-element-newline': 'off',

    // Description: Require parentheses around arrow function arguments
    // Description RU: Требуется заключать аргументы функции arrow в круглые скобки
    // URL: https://eslint.style/rules/default/arrow-parens
    'stylistic/arrow-parens': 'off',

    // Description: Enforce consistent spacing before and after the arrow in arrow functions
    // Description RU: Применяет согласованный интервал до и после стрелки в функциях arrow
    // URL: https://eslint.style/rules/default/arrow-spacing
    'stylistic/arrow-spacing': 'off',

    // Description: Disallow or enforce spaces inside of blocks after opening block and before closing block
    // Description RU: Запрещает или применяет пробелы внутри блоков после открытия блока и перед закрытием блока
    // URL: https://eslint.style/rules/default/block-spacing
    'stylistic/block-spacing': 'off',

    // Description: Enforce consistent brace style for blocks
    // Description RU: Применение согласованного стиля фигурных скобок для блоков
    // URL: https://eslint.style/rules/default/brace-style
    'stylistic/brace-style': 'off',

    // Description: Require or disallow trailing commas
    // Description RU: Требуется или запрещается ставить запятые в конце текста
    // URL: https://eslint.style/rules/default/comma-dangle
    'stylistic/comma-dangle': 'off',

    // Description: Enforce consistent spacing before and after commas
    // Description RU: Необходимо использовать одинаковые интервалы между запятыми и после них
    // URL: https://eslint.style/rules/default/comma-spacing
    'stylistic/comma-spacing': 'off',

    // Description: Enforce consistent comma style
    // Description RU: Применение согласованного стиля с запятой
    // URL: https://eslint.style/rules/default/comma-style
    'stylistic/comma-style': 'off',

    // Description: Enforce consistent spacing inside computed property brackets
    // Description RU: Применение согласованного интервала внутри скобок вычисляемых свойств
    // URL: https://eslint.style/rules/default/computed-property-spacing
    'stylistic/computed-property-spacing': 'off',

    // Description: Enforce consistent line breaks after opening and before closing braces
    // Description RU: Принудительное выполнение последовательных переносов строк после открывающих и закрывающих фигурных скобок
    // URL: https://eslint.style/rules/default/curly-newline
    'stylistic/curly-newline': 'off',

    // Description: Enforce consistent newlines before and after dots
    // Description RU: Необходимо обеспечить согласованность перевода строк до и после точек
    // URL: https://eslint.style/rules/default/dot-location
    'stylistic/dot-location': 'off',

    // Description: Require or disallow newline at the end of files
    // Description RU: Требует или запрещает перевод строки в конце файлов
    // URL: https://eslint.style/rules/default/eol-last
    'stylistic/eol-last': 'off',

    // Description: Require or disallow spacing between function identifiers and their invocations
    // Description RU: Требует или запрещает интервалы между идентификаторами функций и их вызовами
    // URL: https://eslint.style/rules/default/func-call-spacing
    'stylistic/func-call-spacing': 'off',

    // Description: Enforce line breaks between arguments of a function call
    // Description RU: Принудительный перенос строк между аргументами при вызове функции
    // URL: https://eslint.style/rules/default/function-call-argument-newline
    'stylistic/function-call-argument-newline': 'off',

    // Description: Require or disallow spacing between function identifiers and their invocations
    // Description RU: Требуется или запрещается устанавливать интервалы между идентификаторами функций и их вызовами
    // URL: https://eslint.style/rules/default/function-call-spacing
    'stylistic/function-call-spacing': 'off',

    // Description: Enforce consistent line breaks inside function parentheses
    // Description RU: Принудительное выполнение последовательных переносов строк внутри функциональных скобок
    // URL: https://eslint.style/rules/default/function-paren-newline
    'stylistic/function-paren-newline': 'off',

    // Description: Enforce consistent spacing around `*` operators in generator functions
    // Description RU: Обеспечивает соблюдение согласованного интервала между операторами `*` в функциях-генераторах
    // URL: https://eslint.style/rules/default/generator-star-spacing
    'stylistic/generator-star-spacing': 'off',

    // Description: Enforce the location of arrow function bodies
    // Description RU: Обеспечивает соблюдение расположения элементов функций со стрелками
    // URL: https://eslint.style/rules/default/implicit-arrow-linebreak
    'stylistic/implicit-arrow-linebreak': 'off',

    // Description: Enforce consistent indentation
    // Description RU: Применение согласованного отступа
    // URL: https://eslint.style/rules/default/indent
    'stylistic/indent': 'off',

    // Description: Indentation for binary operators
    // Description RU: Отступ для бинарных операторов
    // URL: https://eslint.style/rules/default/indent-binary-ops
    'stylistic/indent-binary-ops': 'off',

    // Description: Enforce or disallow spaces inside of curly braces in JSX attributes and expressions
    // Description RU: Использование или запрет пробелов внутри фигурных скобок в атрибутах и выражениях JSX
    // URL: https://eslint.style/rules/default/jsx-child-element-spacing
    'stylistic/jsx-child-element-spacing': 'off',

    // Description: Enforce closing bracket location in JSX
    // Description RU: Принудительное расположение закрывающей скобки в JSX
    // URL: https://eslint.style/rules/default/jsx-closing-bracket-location
    'stylistic/jsx-closing-bracket-location': 'off',

    // Description: Enforce closing tag location for multiline JSX
    // Description RU: Принудительное расположение закрывающего тега для многострочного JSX
    // URL: https://eslint.style/rules/default/jsx-closing-tag-location
    'stylistic/jsx-closing-tag-location': 'off',

    // Description: Disallow unnecessary JSX expressions when literals alone are sufficient or enforce JSX expressions on literals in JSX children or attributes
    // Description RU: Запрещает использование ненужных выражений JSX, когда достаточно одних литералов, или применяет выражения JSX к литералам в дочерних элементах или атрибутах JSX
    // URL: https://eslint.style/rules/default/jsx-curly-brace-presence
    'stylistic/jsx-curly-brace-presence': 'off',

    // Description: Enforce consistent linebreaks in curly braces in JSX attributes and expressions
    // Description RU: Принудительное использование последовательных переносов строк в фигурных скобках в атрибутах и выражениях JSX
    // URL: https://eslint.style/rules/default/jsx-curly-newline
    'stylistic/jsx-curly-newline': 'off',

    // Description: Enforce or disallow spaces inside of curly braces in JSX attributes and expressions
    // Description RU: Применение или запрет пробелов внутри фигурных скобок в атрибутах и выражениях JSX
    // URL: https://eslint.style/rules/default/jsx-curly-spacing
    'stylistic/jsx-curly-spacing': 'off',

    // Description: Enforce or disallow spaces around equal signs in JSX attributes
    // Description RU: Применение или запрет пробелов вокруг знаков равенства в атрибутах JSX
    // URL: https://eslint.style/rules/default/jsx-equals-spacing
    'stylistic/jsx-equals-spacing': 'off',

    // Description: Enforce proper position of the first property in JSX
    // Description RU: Необходимо обеспечить правильное расположение первого свойства в JSX
    // URL: https://eslint.style/rules/default/jsx-first-prop-new-line
    'stylistic/jsx-first-prop-new-line': 'off',

    // Description: Enforce line breaks before and after JSX elements when they are used as arguments to a function.
    // Description RU: Принудительное использование переносов строк до и после элементов JSX, когда они используются в качестве аргументов функции.
    // URL: https://eslint.style/rules/default/jsx-function-call-newline
    'stylistic/jsx-function-call-newline': 'off',

    // Description: Enforce JSX indentation. Deprecated, use `indent` rule instead.
    // Description RU: Принудительное использование отступов в JSX. Не рекомендуется, вместо этого используйте правило 'отступ'.
    // URL: https://eslint.style/rules/default/jsx-indent
    'stylistic/jsx-indent': 'off',

    // Description: Enforce props indentation in JSX
    // Description RU: Принудительное использование отступов для реквизитов в JSX
    // URL: https://eslint.style/rules/default/jsx-indent-props
    'stylistic/jsx-indent-props': 'off',

    // Description: Enforce maximum of props on a single line in JSX
    // Description RU: Принудительное использование максимального количества реквизитов в одной строке в JSX
    // URL: https://eslint.style/rules/default/jsx-max-props-per-line
    'stylistic/jsx-max-props-per-line': 'off',

    // Description: Require or prevent a new line after jsx elements and expressions.
    // Description RU: Требует или запрещает создание новой строки после элементов и выражений jsx.
    // URL: https://eslint.style/rules/default/jsx-newline
    'stylistic/jsx-newline': 'off',

    // Description: Require one JSX element per line
    // Description RU: Требуется один элемент JSX в строке
    // URL: https://eslint.style/rules/default/jsx-one-expression-per-line
    'stylistic/jsx-one-expression-per-line': 'off',

    // Description: Enforce PascalCase for user-defined JSX components
    // Description RU: Принудительное использование PascalCase для пользовательских компонентов JSX
    // URL: https://eslint.style/rules/default/jsx-pascal-case
    'stylistic/jsx-pascal-case': 'off',

    // Description: Disallow multiple spaces between inline JSX props
    // Description RU: Запрещает использование нескольких пробелов между встроенными реквизитами JSX
    // URL: https://eslint.style/rules/default/jsx-props-no-multi-spaces
    'stylistic/jsx-props-no-multi-spaces': 'off',

    // Description: Enforce the consistent use of either double or single quotes in JSX attributes
    // Description RU: Необходимо обеспечить согласованное использование двойных или одинарных кавычек в атрибутах JSX
    // URL: https://eslint.style/rules/default/jsx-quotes
    'stylistic/jsx-quotes': 'off',

    // Description: Disallow extra closing tags for components without children
    // Description RU: Запретить использование дополнительных закрывающих тегов для компонентов без дочерних элементов
    // URL: https://eslint.style/rules/default/jsx-self-closing-comp
    'stylistic/jsx-self-closing-comp': 'off',

    // Description: Enforce props alphabetical sorting
    // Description RU: Принудительная сортировка реквизитов по алфавиту
    // URL: https://eslint.style/rules/default/jsx-sort-props
    'stylistic/jsx-sort-props': 'off',

    // Description: Enforce whitespace in and around the JSX opening and closing brackets
    // Description RU: Необходимо использовать пробелы в открывающих и закрывающих скобках JSX и вокруг них
    // URL: https://eslint.style/rules/default/jsx-tag-spacing
    'stylistic/jsx-tag-spacing': 'off',

    // Description: Disallow missing parentheses around multiline JSX
    // Description RU: Не допускается использование пропущенных круглых скобок вокруг многострочного текста JSX
    // URL: https://eslint.style/rules/default/jsx-wrap-multilines
    'stylistic/jsx-wrap-multilines': 'off',

    // Description: Enforce consistent spacing between property names and type annotations in types and interfaces
    // Description RU: Обеспечение единообразия интервалов между названиями свойств и аннотациями типов в типах и интерфейсах
    // URL: https://eslint.style/rules/default/key-spacing
    'stylistic/key-spacing': 'off',

    // Description: Enforce consistent spacing before and after keywords
    // Description RU: Обеспечение единообразия интервалов до и после ключевых слов
    // URL: https://eslint.style/rules/default/keyword-spacing
    'stylistic/keyword-spacing': 'off',

    // Description: Enforce position of line comments
    // Description RU: Принудительное расположение комментариев к строке
    // URL: https://eslint.style/rules/default/line-comment-position
    'stylistic/line-comment-position': 'off',

    // Description: Enforce consistent linebreak style
    // Description RU: Принудительное соблюдение согласованного стиля перевода строки
    // URL: https://eslint.style/rules/default/linebreak-style
    'stylistic/linebreak-style': 'off',

    // Description: Require empty lines around comments
    // Description RU: Требует наличия пустых строк вокруг комментариев
    // URL: https://eslint.style/rules/default/lines-around-comment
    'stylistic/lines-around-comment': 'off',

    // Description: Require or disallow an empty line between class members
    // Description RU: Требует или запрещает использование пустой строки между членами класса
    // URL: https://eslint.style/rules/default/lines-between-class-members
    'stylistic/lines-between-class-members': 'off',

    // Description: Enforce a maximum line length
    // Description RU: Установите максимальную длину строки
    // URL: https://eslint.style/rules/default/max-len
    'stylistic/max-len': 'off',

    // Description: Enforce a maximum number of statements allowed per line
    // Description RU: Установите максимальное количество разрешенных выражений в строке
    // URL: https://eslint.style/rules/default/max-statements-per-line
    'stylistic/max-statements-per-line': 'off',

    // Description: Require a specific member delimiter style for interfaces and type literals
    // Description RU: Требуется определенный стиль разделителя элементов для интерфейсов и литералов типов
    // URL: https://eslint.style/rules/default/member-delimiter-style
    'stylistic/member-delimiter-style': 'off',

    // Description: Enforce a particular style for multiline comments
    // Description RU: Применение определенного стиля для многострочных комментариев
    // URL: https://eslint.style/rules/default/multiline-comment-style
    'stylistic/multiline-comment-style': 'off',

    // Description: Enforce newlines between operands of ternary expressions
    // Description RU: Принудительный перевод строк между операндами троичных выражений
    // URL: https://eslint.style/rules/default/multiline-ternary
    'stylistic/multiline-ternary': 'off',

    // Description: Enforce or disallow parentheses when invoking a constructor with no arguments
    // Description RU: Принудительный или запрещающий ввод круглых скобок при вызове конструктора без аргументов
    // URL: https://eslint.style/rules/default/new-parens
    'stylistic/new-parens': 'off',

    // Description: Require a newline after each call in a method chain
    // Description RU: Требуется новая строка после каждого вызова в цепочке методов
    // URL: https://eslint.style/rules/default/newline-per-chained-call
    'stylistic/newline-per-chained-call': 'off',

    // Description: Disallow arrow functions where they could be confused with comparisons
    // Description RU: Запрещает функции со стрелками, если их можно спутать со сравнениями
    // URL: https://eslint.style/rules/default/no-confusing-arrow
    'stylistic/no-confusing-arrow': 'off',

    // Description: Disallow unnecessary parentheses
    // Description RU: Не ставьте лишних круглых скобок
    // URL: https://eslint.style/rules/default/no-extra-parens
    'stylistic/no-extra-parens': 'off',

    // Description: Disallow unnecessary semicolons
    // Description RU: Не ставьте лишних точек с запятой
    // URL: https://eslint.style/rules/default/no-extra-semi
    'stylistic/no-extra-semi': 'off',

    // Description: Disallow leading or trailing decimal points in numeric literals
    // Description RU: Запрещает использование начальных и конечных десятичных знаков в числовых литералах
    // URL: https://eslint.style/rules/default/no-floating-decimal
    'stylistic/no-floating-decimal': 'off',

    // Description: Disallow mixed binary operators
    // Description RU: Запрещает смешанные двоичные операторы
    // URL: https://eslint.style/rules/default/no-mixed-operators
    'stylistic/no-mixed-operators': 'off',

    // Description: Disallow mixed spaces and tabs for indentation
    // Description RU: Не допускать смешанные пробелы и символы табуляции в отступах
    // URL: https://eslint.style/rules/default/no-mixed-spaces-and-tabs
    'stylistic/no-mixed-spaces-and-tabs': 'off',

    // Description: Disallow multiple spaces
    // Description RU: Запрещает использование нескольких пробелов
    // URL: https://eslint.style/rules/default/no-multi-spaces
    'stylistic/no-multi-spaces': 'off',

    // Description: Disallow multiple empty lines
    // Description RU: Запрещает использование нескольких пустых строк
    // URL: https://eslint.style/rules/default/no-multiple-empty-lines
    'stylistic/no-multiple-empty-lines': 'off',

    // Description: Disallow all tabs
    // Description RU: Запретить все вкладки
    // URL: https://eslint.style/rules/default/no-tabs
    'stylistic/no-tabs': 'off',

    // Description: Disallow trailing whitespace at the end of lines
    // Description RU: Запрет на использование завершающих пробелов в конце строк
    // URL: https://eslint.style/rules/default/no-trailing-spaces
    'stylistic/no-trailing-spaces': 'off',

    // Description: Disallow whitespace before properties
    // Description RU: Запрет на использование пробелов перед свойствами
    // URL: https://eslint.style/rules/default/no-whitespace-before-property
    'stylistic/no-whitespace-before-property': 'off',

    // Description: Enforce the location of single-line statements
    // Description RU: Принудительное использование расположения однострочных инструкций
    // URL: https://eslint.style/rules/default/nonblock-statement-body-position
    'stylistic/nonblock-statement-body-position': 'off',

    // Description: Enforce consistent line breaks after opening and before closing braces
    // Description RU: Принудительно выполнять последовательные переносы строк после открытия и перед закрытием фигурных скобок
    // URL: https://eslint.style/rules/default/object-curly-newline
    'stylistic/object-curly-newline': 'off',

    // Description: Enforce consistent spacing inside braces
    // Description RU: Обеспечение единообразия интервалов внутри фигурных скобок
    // URL: https://eslint.style/rules/default/object-curly-spacing
    'stylistic/object-curly-spacing': 'off',

    // Description: Enforce placing object properties on separate lines
    // Description RU: Принудительное размещение свойств объектов в отдельных строках
    // URL: https://eslint.style/rules/default/object-property-newline
    'stylistic/object-property-newline': 'off',

    // Description: Require or disallow newlines around variable declarations
    // Description RU: Требует или запрещает перевод строк вокруг объявлений переменных
    // URL: https://eslint.style/rules/default/one-var-declaration-per-line
    'stylistic/one-var-declaration-per-line': 'off',

    // Description: Enforce consistent linebreak style for operators
    // Description RU: Применение согласованного стиля перевода строки для операторов
    // URL: https://eslint.style/rules/default/operator-linebreak
    'stylistic/operator-linebreak': 'off',

    // Description: Require or disallow padding within blocks
    // Description RU: Требует или запрещает заполнение внутри блоков
    // URL: https://eslint.style/rules/default/padded-blocks
    'stylistic/padded-blocks': 'off',

    // Description: Require or disallow padding lines between statements
    // Description RU: Требует или запрещает заполнение строк между операторами
    // URL: https://eslint.style/rules/default/padding-line-between-statements
    'stylistic/padding-line-between-statements': 'off',

    // Description: Require quotes around object literal, type literal, interfaces and enums property names
    // Description RU: Требуется заключать в кавычки объектный литерал, литерал типа, имена свойств интерфейсов и перечислений
    // URL: https://eslint.style/rules/default/quote-props
    'stylistic/quote-props': 'off',

    // Description: Enforce the consistent use of either backticks, double, or single quotes
    // Description RU: Обеспечить согласованное использование обратных кавычек, двойных или одинарных кавычек
    // URL: https://eslint.style/rules/default/quotes
    'stylistic/quotes': 'off',

    // Description: Enforce spacing between rest and spread operators and their expressions
    // Description RU: Обязательный интервал между операторами rest и spread и их выражениями
    // URL: https://eslint.style/rules/default/rest-spread-spacing
    'stylistic/rest-spread-spacing': 'off',

    // Description: Require or disallow semicolons instead of ASI
    // Description RU: Требуется или запрещается использовать точку с запятой вместо ASI
    // URL: https://eslint.style/rules/default/semi
    'stylistic/semi': 'off',

    // Description: Enforce consistent spacing before and after semicolons
    // Description RU: Принудительно использовать согласованный интервал до и после точки с запятой
    // URL: https://eslint.style/rules/default/semi-spacing
    'stylistic/semi-spacing': 'off',

    // Description: Enforce location of semicolons
    // Description RU: Принудительно использовать расположение точек с запятой
    // URL: https://eslint.style/rules/default/semi-style
    'stylistic/semi-style': 'off',

    // Description: Enforce consistent spacing before blocks
    // Description RU: Необходимо обеспечить согласованный интервал между блоками
    // URL: https://eslint.style/rules/default/space-before-blocks
    'stylistic/space-before-blocks': 'off',

    // Description: Enforce consistent spacing before function parenthesis
    // Description RU: Необходимо обеспечить согласованный интервал между функциональными скобками
    // URL: https://eslint.style/rules/default/space-before-function-paren
    'stylistic/space-before-function-paren': 'off',

    // Description: Enforce consistent spacing inside parentheses
    // Description RU: Требуется соблюдение согласованного интервала внутри круглых скобок
    // URL: https://eslint.style/rules/default/space-in-parens
    'stylistic/space-in-parens': 'off',

    // Description: Require spacing around infix operators
    // Description RU: Требуется использовать интервал между инфиксными операторами
    // URL: https://eslint.style/rules/default/space-infix-ops
    'stylistic/space-infix-ops': 'off',

    // Description: Enforce consistent spacing before or after unary operators
    // Description RU: Обеспечение соблюдения согласованного интервала до или после унарных операторов
    // URL: https://eslint.style/rules/default/space-unary-ops
    'stylistic/space-unary-ops': 'off',

    // Description: Enforce consistent spacing after the `//` or `/*` in a comment
    // Description RU: Используйте согласованный интервал после '//' или '/*' в комментарии
    // URL: https://eslint.style/rules/default/spaced-comment
    'stylistic/spaced-comment': 'off',

    // Description: Enforce spacing around colons of switch statements
    // Description RU: Обязательный интервал между двоеточиями в операторах switch
    // URL: https://eslint.style/rules/default/switch-colon-spacing
    'stylistic/switch-colon-spacing': 'off',

    // Description: Require or disallow spacing around embedded expressions of template strings
    // Description RU: Требует или запрещает интервал между встроенными выражениями строк шаблона
    // URL: https://eslint.style/rules/default/template-curly-spacing
    'stylistic/template-curly-spacing': 'off',

    // Description: Require or disallow spacing between template tags and their literals
    // Description RU: Требуется или запрещается устанавливать интервал между тегами шаблона и их литералами
    // URL: https://eslint.style/rules/default/template-tag-spacing
    'stylistic/template-tag-spacing': 'off',

    // Description: Require consistent spacing around type annotations
    // Description RU: Требуется согласованный интервал между аннотациями типа
    // URL: https://eslint.style/rules/default/type-annotation-spacing
    'stylistic/type-annotation-spacing': 'off',

    // Description: Enforces consistent spacing inside TypeScript type generics
    // Description RU: Обеспечивает единообразие интервалов внутри типовых символов TypeScript
    // URL: https://eslint.style/rules/default/type-generic-spacing
    'stylistic/type-generic-spacing': 'off',

    // Description: Expect space before the type declaration in the named tuple
    // Description RU: Перед объявлением типа в именованном кортеже должен быть пробел
    // URL: https://eslint.style/rules/default/type-named-tuple-spacing
    'stylistic/type-named-tuple-spacing': 'off',

    // Description: Require parentheses around immediate `function` invocations
    // Description RU: Требуется заключать в круглые скобки непосредственные вызовы функций
    // URL: https://eslint.style/rules/default/wrap-iife
    'stylistic/wrap-iife': 'off',

    // Description: Require parenthesis around regex literals
    // Description RU: Требуется заключать литералы регулярных выражений в круглые скобки
    // URL: https://eslint.style/rules/default/wrap-regex
    'stylistic/wrap-regex': 'off',

    // Description: Require or disallow spacing around the `*` in `yield*` expressions
    // Description RU: Требуется или запрещается устанавливать интервал между символами '*' в выражениях 'yield*'
    // URL: https://eslint.style/rules/default/yield-star-spacing
    'stylistic/yield-star-spacing': 'off',
  }
}
